test/streams-test.js

```diff
@@ -331,3 +329,54 @@ describe("Streaming sugar", function() {
     });
 });
 
+describe("Encoding using internal modules with surrogates in separate chunks:", function () {
+    it("a single string", checkEncodeStream({
+        encoding: "utf8",
+        input: ["\uD83D\uDE3B"]
+    }))
```

Comment:
could you explain why you've decided to get back to calling checkEncodeStream directly? Did my previous suggestion not work?